### PR TITLE
RAINCATCH-1292 - extending UI interface

### PR DESCRIPTION
**Pull request by wtrocki**
Guide for extending workorder UI. 
Angular specific. 
I decided to move out from procedure as there will be so many different flows and variations depending on the use case.

## Notes

We can reuse here screenshots from guide that is in WIP by @austincunningham to show example extension at later stage.

### CODE REVIEW

File: docs/workforce-management-framework/topics/con-wfm-extending-user-interface.adoc [hunk 1]
```diff
@@ -0,0 +1,56 @@
+= Extending {WFM-RC-NameShort} details
+
+RainCatcher framework provides set of extension points that allows developers to modify angular user interface. All modifications can be done by supplying additional module that will provide set of the angular.js directives used to extend mobile and portal user interface.
+
+Demo application is being shipped with example `angularjs-extensions module` that extends demo UI:
+
```
Comment (austincunningham) [inline docs/workforce-management-framework/topics/con-wfm-extending-user-interface.adoc]: Try not to use Demo

File: docs/workforce-management-framework/topics/con-wfm-extending-user-interface.adoc [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+ - `workorder-data-edit`: Data-edit directive, should implement fields that allow to change/edit data.
+
+Directives should operate on existing field outside workorder to avoid name conflicts:
+`workorder.data.yourcustomfield`
+
+== User interface
+
+If your user interface requires additional styles or libraries they need to be included in top level application.Standard bootstrap and angular material directives are available in top level mobile and portal applications.
+
+https://material.angularjs.org/latest/api
```
Comment (austincunningham) [inline docs/workforce-management-framework/topics/con-wfm-extending-user-interface.adoc]: URL lacks context. Maybe add 'For reference see '

File: docs/workforce-management-framework/topics/con-extending-the-user-interface.adoc [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+|`workorder-data`
+|A Data directive used to implement static fields that display data.
+
+|`workorder-data-edit`
+|A Data-edit directive used to implement fields that allow modifications to be made to a _workorder's_ data.
+
+|===
+
+NOTE: Directives are to operate (with respect to existing fields) outside of a _workorder_ to avoid naming conflicts, for example, `workorder.data.yourcustomfield`
```
Comment (jstaffor) [inline docs/workforce-management-framework/topics/con-extending-the-user-interface.adoc]: @wtrocki - can you take a look at this and make sure I haven't lost the technical context as a result of my change?

File: docs/workforce-management-framework/topics/con-extending-the-user-interface.adoc [hunk 4]
```diff
@@ -0,0 +1,57 @@
+[id='{context}-con-extending-the-user-interface']
+= Extending the User Interface
```
Comment (jstaffor) [inline docs/workforce-management-framework/topics/con-extending-the-user-interface.adoc]: @wtrocki - "Extending the User Interface" - is this ok or am I missing a more descriptive heading?

**Review by jstaffor — DISMISSED** [review]

File: docs/workforce-management-framework/topics/con-extending-the-user-interface.adoc (hunk 4 above)
Comment (wtrocki) [reply docs/workforce-management-framework/topics/con-extending-the-user-interface.adoc]: To provide more context. We have now only way to extend workorders data, but I wanted to create general space where all UI extensions will land in the future. Description is ok. 
If we want to extend that we can say `extending angular user interface`

**Comment by wtrocki** [comment]
@jstaffor - Original PR contained only single one file with the changes. All the changes should be contained in the single files.

After review PR containing 30 additional files that have nothing to do with the original purpose of this PR. Additionally file was changed in the way that I do not see exact changes (Entire file shows as new). This is making this review absolutely impossible and really inconvenient. 
PR also cannot be merged. 

I'm going to fix that, but let's avoid some merge commits and conflicts. 
For single file you can edit changes directly in the github without worrying about git commads.

File: docs/workforce-management-framework/topics/con-wfm-extending-user-interface.adoc [hunk 5]
```diff
@@ -0,0 +1,57 @@
+[id='{context}-con-extending-the-user-interface']
+= Extending the User Interface
+
+{Feature} framework provides a set of extension points which allows a developer to modify an Angular User Interface (UI).
+All modifications are done by supplying additional modules which provide a set of the Angular.js directives and are used to extend the Mobile and Portal UI.
+The {Feature} Application is shipped with the example `angularjs-extensions` example that extends the Demo UI:
```
Comment (wtrocki) [inline docs/workforce-management-framework/topics/con-wfm-extending-user-interface.adoc]: Mobile and portal applications

File: docs/workforce-management-framework/topics/con-wfm-extending-user-interface.adoc [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+ * https://github.com/feedhenry-raincatcher/raincatcher-angularjs/tree/master/packages/angularjs-extensions
+
+This Module contains the example directives that are used for demo purposes and it can be used as a template to extend the existing {Feature} UI.
+
+== Extending Workorders
+
+By default, {Feature} provides a minimal set of defined fields for a _workorder_.
+Depending on the needs of the business, a developer can add additional data and UI elements, for example, gps location, an address or a description UI element.
+
+=== Building Workorders
```
Comment (wtrocki) [inline docs/workforce-management-framework/topics/con-wfm-extending-user-interface.adoc]: This is very good example when meaning is changed and no longer makes sense.
Extending Workorders != Budiling Workorders. Can we revert?

We mention extending bellow.  If we do not like extending we can use something like: 
Adding metadata etc.

**Comment by wtrocki** [comment]
@jstaffor Really god job! Content makes sense. Just 2 minor comments. Apart from that is ready to merge.

**Comment by wtrocki** [comment]
@jstaffor Really sorry - just noticed that my rebase reverted file to original name. That was mistake. Feel free to correct that as well.

File: docs/workforce-management-framework/topics/con-wfm-extending-user-interface.adoc (hunk 5 above)
Comment (jstaffor) [reply docs/workforce-management-framework/topics/con-wfm-extending-user-interface.adoc]: Will add.

File: docs/workforce-management-framework/topics/con-wfm-extending-user-interface.adoc (hunk 6 above)
Comment (jstaffor) [reply docs/workforce-management-framework/topics/con-wfm-extending-user-interface.adoc]: I will revert this back to origin "=== Building"

File: docs/workforce-management-framework/topics/con-wfm-extending-user-interface.adoc [hunk 7]
```diff
@@ -14,7 +14,7 @@ This Module contains the example directives that are used for demo purposes and
 By default, {Feature} provides a minimal set of defined fields for a _workorder_.
 Depending on the needs of the business, a developer can add additional data and UI elements, for example, gps location, an address or a description UI element.
 
-=== Building Workorders
+=== Building
```
Comment (wtrocki) [inline docs/workforce-management-framework/topics/con-wfm-extending-user-interface.adoc]: Once again - extending not building :)

Comment (wtrocki) [reply docs/workforce-management-framework/topics/con-wfm-extending-user-interface.adoc]: I have added - building directives. Should be ok.

**Comment by wtrocki** [comment]
Great job! It looks nice!

File: docs/workforce-management-framework/topics/con-wfm-extending-user-interface.adoc (hunk 7 above)
Comment (jstaffor) [reply docs/workforce-management-framework/topics/con-wfm-extending-user-interface.adoc]: @wtrocki - can you take a look back though the history here - I think you are confused and you are, being honest,  causing me to become _very_ confused and I don't appreciate it. 

1) This heading was originally : "=== Building"
2) I changed it to : "=== Building Workorders"
3) You made the comment: "Extending Workorders != Budiling Workorders. Can we revert?"
4) I reverted it to original, as per your comment: "=== Building"
5) You then add the comment: "Once again - extending not building :)"
6) You then change it to: "=== Building directives"

Comment (wtrocki) [reply docs/workforce-management-framework/topics/con-wfm-extending-user-interface.adoc]: I think that git changes got us confused - myself especially. 
TLDR;

- Building was ok but were incomplete
- Building workorder were missleading as were building extension directives here.
- Extending workorder will be better but.. it's already mentioned on the top
- Building directives - was my final candidate as our target was to extend this to provide info what we going to build.
made change and merged that.
